src/main/scala/de/unruh/isabelle/pure/TheoryHeader.scala

```diff
@@ -27,11 +31,24 @@ final class TheoryHeader private[TheoryHeader] (val mlValue: MLValue[TheoryHeade
 object TheoryHeader extends MLValueWrapper.Companion[TheoryHeader] {
   override protected val mlType: String = "Thy_Header.header"
   override protected val predefinedException: String = "E_TheoryHeader"
+
+  /**
+    * Parses a theory header from text.
+    *
+    * The text can be a prefix like "theory Foo imports Bar Baz begin" or a full theory file.
+    */
+  def parse(text: String)(implicit isabelle: Isabelle) : TheoryHeader =
```

Comment:
Maybe rename this to `read`. `parse` is a sensible name, of course, but `read` would keep more of a correspondence with the Isabelle/ML names. This makes it easier to for people who work both in Isabelle/ML and scala-isabelle.

Reply:
Ok